### DISCUSSION TITLE
Questions about newlib library

### DISCUSSION THREAD

**Discussion by chojpsh1** [zephyrproject-rtos/zephyr#52357] [Q&A] [answered]
Hello, I have some questions about newlib math library (libm.a).

I checked that Zephyr 3.x.x version supports the linking libm.a through CONFIG_NEWLIB_LIBC. But, CONFIG_NEWLIB_LIBC config is used to link the libc.a and libm.a together.
I'm looking for a way to link only libm.a. I would like to ask linking only libm.a is possible?
It's my first question and the second is about size.

The size of libc.a and libm.a seems to be large. (I saw zephyr-sdk-0.12.4/aarch64-zephyr-elf/aarch64-zephyr-elf/lib/libc.a and libm.a. And libc.a's size is 1.3MB and libm.a's size is 0.5MB)
If linking only libm.a is impossible, 1.8MB over size will be used for just using math function. And there seems to be no big difference between lib*.a and lib*_nano.a, so I would like to ask building the newlib library smaller is possible?

So to sum up, I have two questions about newlib library.
1. Is it possible to link only libm.a?
2. Is it possible to build the newlib library smaller?

Thank you!

**Comment by stephanosio** [comment, marked as answer]
> Is it possible to link only libm.a?

No.

> Is it possible to build the newlib library smaller?

Not without taking away functionalities.

In general, please use the picolibc if you are working on something footprint critical:
https://docs.zephyrproject.org/latest/kconfig.html#CONFIG_PICOLIBC

**Reply by chojpsh1** [reply]
Thank you for your check!
